types/src/v2/params.rs

```diff
@@ -68,7 +68,90 @@ impl<'a> RpcParams<'a> {
 		Self(raw)
 	}
 
-	/// Attempt to parse all parameters as array or map into type T
+	fn next_inner<T>(&mut self) -> Option<Result<T, CallError>>
+	where
+		T: Deserialize<'a>,
+	{
+		let mut json = self.0?.trim_start();
+
+		match json.as_bytes().get(0)? {
+			b']' => {
+				self.0 = None;
+
+				return None;
+			}
+			b'[' | b',' => json = &json[1..],
```

Comment:
this could panic if the `str slice` is something like `[`, `,`, `[∀`

technically I think, a nasty client could send it erroneous json objects as params and we would accept them?

Reply:
looks like I was wrong

Reply:
Slicing `"["` at offset `1` gives you an empty slice, and there is no issue slicing in front of a non-ascii character since it's at boundary there.

Reply:
yeah right, I guess I'm getting tired of code reading today.